### PR TITLE
Add graphing and filtering to student report

### PR DESCRIPTION
**Pull request by jhilde**
This is the consolidated pull request for phase 2 (graphs) and 3 (filtering) of the new student report.

Supersedes pull request #873 
[student_report.pdf](https://github.com/studentinsights/studentinsights/files/856881/student_report.pdf)

### CODE REVIEW

**Comment by jhilde** [comment]
Pinging on this @alexsoble @kevinrobinson. Let me know if you have feedback.

File: app/controllers/students_controller.rb [hunk 1]
```diff
@@ -144,29 +148,52 @@ def student_feed(student, restricted_notes: false)
...
 
-    # Load last 2 student schools years for absences, tardies, and discipline incidents
-    @student_school_years = @student.student_school_years.includes(:absences).includes(:tardies).includes(:discipline_incidents).where(school_year_id: [prior_school_year_id, current_school_year_id])
+  end
+
+  def set_up_student_report_data
+    @student = Student.find(params[:id])
+    @current_educator = current_educator
+    @url = root_url.chomp('/') + request.path
+    @sections = (params[:sections] || "").split(/\s*,\s*/)
```
Comment (alexsoble) [inline app/controllers/students_controller.rb]: @jhilde Question -- what is the `.split(/\s*,\s*/)` code doing?

File: app/controllers/students_controller.rb [hunk 2]
```diff
@@ -144,29 +148,52 @@ def student_feed(student, restricted_notes: false)
...
+    return school_year_ids
 
-    # Load last 2 student schools years for absences, tardies, and discipline incidents
-    @student_school_years = @student.student_school_years.includes(:absences).includes(:tardies).includes(:discipline_incidents).where(school_year_id: [prior_school_year_id, current_school_year_id])
+  end
+
+  def set_up_student_report_data
+    @student = Student.find(params[:id])
+    @current_educator = current_educator
+    @url = root_url.chomp('/') + request.path
```
Comment (alexsoble) [inline app/controllers/students_controller.rb]: @jhilde Quick question, what is `@url = root_url.chomp('/') + request.path` doing? Thanks!

**Comment by alexsoble** [comment]
Hi @jhilde!  

I left a couple of in-line code comments about specific lines of code I want to understand a little better.

Also posting a GIF of what this new filtering UI looks like, below. I recommend posting screenshots and/or GIFs along with your PRs, it makes it clearer what the code you're merging in does.  

# GIF
new-student-report-ui

What are your thoughts on the UI Justin? 

# Code duplication question

I'm noticing some of of the functions on `student_profile_pdf.js` are the same as functions on `profile_bar_chart.js`: `monthKeys`, `defaultMonthKey`, `eventsToMonthBuckets`. What do you think about extracting these methods out so that both of the files can share the same methods?

Cheers -- hope these questions and feedback are useful. Let me know what other ideas, thoughts, or questions are in your brain!

**Comment by alexsoble** [comment]
P.S.  @jhilde let me know if you want to jump on a screenshare or hangout some time to talk about rebasing! We can use rebasing to get rid of some of the "noisy" commits that say things like `Merge branch 'master' of https://github.com/jhilde/studentinsights` and leave us an easier-to-read history.

File: app/controllers/students_controller.rb (hunk 1 above)
Comment (jhilde) [inline app/controllers/students_controller.rb]: Should be splitting on comma separated values. The sections param is a comma delimited string of section names. It ignores leading and trailing white space which may be irrelevant.

File: app/controllers/students_controller.rb (hunk 2 above)
Comment (jhilde) [inline app/controllers/students_controller.rb]: I realize that's a holdover from @erose's initial take on the report. Initially the url was included in the footer of the report. I removed it to save space.

So we can remove the variable as well.

**Comment by jhilde** [comment]
@alexsoble Thanks for the helpful feedback! Those functions are exactly the same for that purpose. I'd be happy to factor them out. Thoughts about where they should belong?

And yes, I'd love to work through rebase and norms around verbosity on check in comments, etc.

**Comment by jhilde** [comment]
And @alexsoble, what did you use to create the animated gif. Very cool.

**Comment by alexsoble** [comment]
@jhilde LICEcap! http://www.cockos.com/licecap/

**Comment by alexsoble** [comment]
@jhilde Let's figure out a good time to jump on a screenshare over email!

**Comment by kevinrobinson** [comment]
Hi @jhilde, sorry for the delay on reviewing, too much traveling the last two weeks.  And thanks for jumping in @alexsoble so we can keep moving things forward.  🌮 👍

File: app/controllers/students_controller.rb (hunk 2 above)
Comment (alexsoble) [inline app/controllers/students_controller.rb]: @jhilde Great, yeah let's take it out if it's not used.

File: app/controllers/students_controller.rb (hunk 1 above)
Comment (alexsoble) [inline app/controllers/students_controller.rb]: OK cool, yeah I was wondering if `.split(",")` could do the job just as well.

Comment (jhilde) [inline app/controllers/students_controller.rb]: Switched to the simpler .split(",")

File: app/controllers/students_controller.rb (hunk 2 above)
Comment (jhilde) [inline app/controllers/students_controller.rb]: Removed superfluous code.

**Comment by jhilde** [comment]
Moved shared graphing functions into a helper app/assets/javascripts/helpers/graph_helpers.js, removed code from app/assets/javascripts/student_profile/profile_bar_chart.js and pdf/student_profile_pdf.js, reworked the tests.

**Comment by alexsoble** [comment]
Nice job with the rebase @jhilde! Will merge in and push to prod early next week to avoid any Friday late afternoon push-to-prod jinxes!

**Comment by jhilde** [comment]
Awesome! Thanks.

**Comment by alexsoble** [comment]
🚢 ing

**Comment by kevinrobinson** [comment]
💥 awesome @jhilde! 👍

**Pull request merged**